types/asset.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+const FormatUniABSPrefix = "uni:"
+
+// ConvertIdToTokenKeyId return the store key suffix of a token
+func ConvertIdToTokenKeyId(tokenId string) (key string, err error) {
+	var reToken = `[A-Za-z0-9\.]{3,17}`
+	if !regexp.MustCompile(reToken).MatchString(tokenId) {
```

Comment:
It's better to precompile first，as follow:
` 
     reToken  := regexp.MustCompile(`[A-Za-z0-9\.]{3,17})
     reToken. MatchString(tokenId)
`